elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchConnectionSettings.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.elasticsearch
+
+final class ElasticsearchConnectionSettings private (
+    val baseUrl: String,
+    val username: Option[String],
+    val password: Option[String]
+) {
+
+  def withBaseUrl(value: String): ElasticsearchConnectionSettings = copy(baseUrl = value)
+
+  def withCredentials(username: String, password: String): ElasticsearchConnectionSettings =
+    copy(username = Option(username), password = Option(password))
+
+  def hasCredentialsDefined: Boolean = username.isDefined && password.isDefined
+
+  def copy(baseUrl: String = baseUrl,
+           username: Option[String] = username,
+           password: Option[String] = password): ElasticsearchConnectionSettings =
+    new ElasticsearchConnectionSettings(baseUrl = baseUrl, username = username, password = password)
+
+  override def toString =
+    s"""ElasticsearchSourceSettings(baseUrl=$baseUrl,username=$username,password=${password.fold("")(_ => "***")}"""
+}
+
+object ElasticsearchConnectionSettings {
+
+  val Default = new ElasticsearchConnectionSettings(
+    baseUrl = "",
+    username = None,
+    password = None
+  )
+
+  /** Scala API */
+  def apply(): ElasticsearchConnectionSettings = Default
```

Comment:
Require the base URL to be passed in:

```suggestion
  def apply(baseUrl: String) = new ElasticsearchConnectionSettings(baseUrl, None, None)
```

Reply:
This did force me to also remove the Default for ElasticsearchSourceSettings and ElasticsearchWriteSettings since they relied on ElasticsearchConnectionSettings.Default. Are you ok with how it is now?